src/approaches/interactive_learning_approach.py

```diff
@@ -0,0 +1,178 @@
+"""An approach that learns predicates from a teacher.
+"""
+
+import itertools
+from typing import Set, Callable, List, Sequence, Collection
+import numpy as np
+import torch
+from gym.spaces import Box
+
+from predicators.src import utils
+from predicators.src.approaches import TAMPApproach
+from predicators.src.structs import State, Predicate, ParameterizedOption, \
+    Type, Task, Action, Operator, Dataset, GroundAtom, ActionTrajectory, \
+    Object
+from predicators.src.operator_learning import learn_operators_from_data
+from predicators.src.models import MLPClassifier
+from predicators.src.utils import get_object_combinations
+from predicators.src.settings import CFG, get_save_path
+
+
+class InteractiveLearningApproach(TAMPApproach):
+    """An approach that learns predicates from a teacher.
+    """
+    def __init__(self,
+                 simulator: Callable[[State, Action], State],
+                 true_predicates: Set[Predicate],
+                 initial_options: Set[ParameterizedOption],
+                 types: Set[Type],
+                 action_space: Box,
+                 train_tasks: List[Task]) -> None:
+        # Only the teacher is allowed to know about the initial predicates
+        self._known_predicates = {p for p in true_predicates
+                                  if p.name in CFG.interactive_known_predicates}
+        predicates_to_learn = true_predicates - self._known_predicates
+        self._teacher = _Teacher(true_predicates, predicates_to_learn)
+        # All seen data
+        self.dataset: List[ActionTrajectory] = []
```

Comment:
Let's stick with private attributes (names starting with underscores) unless the attribute needs to be used outside the class